### ISSUE TITLE
[Enhancement] Bring Back The Old Effects!

### ISSUE THREAD

**Issue by Axemasta** [xamarin/XamarinCommunityToolkit#483]
## Summary

This repository used to contain a nice variety of effects that i've tended to use in various places. The old NuGet package is still available and I do still use it however it would be nice to carry those effects over to this revitalised version of the lib. 

Below is a list of some of the available effects
![List of old effects](https://user-images.githubusercontent.com/33064621/96876980-fdb01480-1470-11eb-89d1-859188b5ac63.png)

Now I think some of these effects are probably outdated however I would assume not all of them are useless. I specifically want to use `EntryRemoveBorder` and `EntryRemoveLine` in many places as it's very necessary for styling.

Was there a reason that these effects were removed from this lib? a cursory search of the issues doesn't reveal much and I haven't looked at the commit history.

Since the old NuGet package is pointing to this repository ([link](https://www.nuget.org/packages/Xamarin.Toolkit.Effects)), I think the effects should either be added back or it should be made clear they are no longer available in this lib.

## Who Will Do The Work?

- [X] I am willing to take this on myself
- [ ] Just putting this out there for someone to pick up

**Comment by AndreiMisiukevich** [comment]
Don't you mind to refactor the code a little bit?
At least names :)

**Comment by jfversluis** [comment]
I'm not opposed to bring them back, but I would want to check which are actually still needed. Maybe some of them are now supported by Forms out of the box? If you can update or comment here which ones you actually want to bring back, what they actually do and give us some more details, that would be cool. Thanks!

**Comment by Axemasta** [comment]
Cool, I couldnt find any docs for these old effects so I've had a look at them and wildy assumed their functionality. here is a table of the old effects, what they do and if they are supported by Forms [OTB](https://www.google.com/search?q=what+does+out+fo+the+box+mean&oq=what+does+out+fo+the+box+mean&aqs=chrome..69i57j0i13i457j0i13j0i10i13i30j0i13i30l4.3194j1j4&sourceid=chrome&ie=UTF-8)

| Effect                    | Purpose                                                      | Supported Out The Box?          |
| ------------------------- | ------------------------------------------------------------ | ------------------------------- |
| EntryCapitalizeKeyboard   | Capitalize all inputs on an `Entry`                          | [ ]                             |
| EntryClear                | Adds clear button to `Entry`                                 | [X] - `ClearButtonVisibility`   |
| EntryDisableAutoCorrect   | Disables autocorrect for `Entry`                             | [ ]                             |
| EntryItalicPlaceholder    | Makes `Entry` placeholder italic                             | [X] - `FontAttributes="Italic"` |
| EntryRemoveBorder         | Removes `Entry` border on iOS                                | [ ]                             |
| EntryRemoveLine           | Removes `Entry` underline on Android                         | [ ]                             |
| EntrySelectAllText        | Selects all text in entry when focussed                      | [ ]                             |
| LabelCustomFont           | Allows a custom font for `Label`                             | [X] - `FontFamily`              |
| LabelMultiLine            | Specify the amount of lines a `Label` can have               | [X] - `MaxLines`                |
| LabelSizeFontToFit        | I assume this sizes the label font to fit the available space however it didn't do anything in my sample project | ?                               |
| PickerChangeColorEffect   | No idea what this did + it crashed my app when I tried to use the effect | ?                               |
| SearchBarSuggestionEffect | I didn't try this one                                        | ?                               |
| SwitchChangeColorEffect   | Again no idea what this one does, attached it to my switch and it didn't look any different to the original | ?                               |
| ViewBlurEffect            | This one crashed my sample app                               | ?                               |



The effects I'm personally interested are the entry remove lines / borders but I think the following effects would be useful for developers:

- EntryCapitalizeKeyboard
- EntryDisableAutoCorrect
- EntryRemoveBorder
- EntryRemoveLine
- EntrySelectAllText

We could potentially combine the **EntryRemoveBorder** and **EntryRemoveLine** effects into one single effect, thats what I'd do.

Anyway what are your thoughts? would people be happy to see these effects added back?

**Comment by jfversluis** [comment]
Wooow thanks for this @Axemasta I think autosuggest and capitalization can be done with this? https://docs.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/text/entry#customize-the-keyboard

**Comment by Axemasta** [comment]
Oh excellent, 2 more to strike off the list. I'll make a PR tomorrow with

- EntryRemoveBorder
- EntryRemoveLine
- EntrySelectAllText

**Comment by AndreiMisiukevich** [comment]
@Axemasta 

* EntryNoBorderEffect
* EntryNoUnderlineEffect
* EntrySelectAllTextEffect

I have a question. Does it make sense to combine EntryNoBorderEffect & EntryNoUnderlineEffect into a single effect?
Seems the first one is only for iOS and the second one is for Android.

```csharp
class EntryBorderEffect
{
    public bool HasBorder { get; set; }
    public bool HasUnderline { get; set; }
}
```

Please keep in mind that it is not a strict request. That's more just a question 👍

**Comment by Axemasta** [comment]
@AndreiMisiukevich I think this is a good idea, I actually mentioned the same thing in my original post. In my own apps I have 1 effect that does both on each platform, I wonder if they just created them separately and then threw them into the repo.

**Comment by AlleSchonWeg** [comment]
I need the LabelSizeFontToFit-Effect. 🙈

**Comment by Axemasta** [comment]
Merged in #492
